driver/src/solution_submission/mod.rs

```diff
@@ -120,48 +128,193 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         solution: Solution,
         claimed_objective_value: U256,
     ) -> Result<(), SolutionSubmissionError> {
-        self.contract
-            .submit_solution(
-                batch_index.clone(),
-                solution.clone(),
-                claimed_objective_value,
+        retry_with_gas_price_increase(
+            self.contract,
+            batch_index,
+            solution.clone(),
+            claimed_objective_value,
+            self.gas_price_estimating,
+            60_000_000_000u64.into(),
+        )
+        .map_err(|err| {
+            extract_transaction_receipt(&err)
+                .and_then(|tx| {
+                    let block_number = tx.block_number?;
+                    match self.contract.get_solution_objective_value(
+                        batch_index,
+                        solution,
+                        Some(block_number.into()),
+                    ) {
+                        Ok(_) => None,
+                        Err(e) => Some(SolutionSubmissionError::from(e)),
+                    }
+                })
+                .unwrap_or_else(|| SolutionSubmissionError::Unexpected(err.into()))
+        })
+        .map(|_| ())
+    }
+}
+
+fn retry_with_gas_price_increase(
+    contract: &dyn StableXContract,
+    batch_index: U256,
+    solution: Solution,
+    claimed_objective_value: U256,
+    gas_price_estimating: &dyn GasPriceEstimating,
+    gas_cap: U256,
+) -> Result<(), MethodError> {
+    const INCREASE_FACTOR: u32 = 2;
+    const BLOCK_TIMEOUT: usize = 1;
+    const DEFAULT_GAS_PRICE: u64 = 15_000_000_000;
+
+    let mut gas_price = match gas_price_estimating.estimate_gas_price() {
```

Comment:
Do you have any thoughts if this be done as well inside each loop in case the gas price changed in the meantime?

Reply:
Do you think the gas price could double in 2 blocks?

Reply:
I implemented this, which I guess will help when the gas price is increasing rapidly

Reply:
Unfortunately I didn't catch this at the time. I meant to only make the estimation once. Not exactly sure how this slipped by.

Reply:
Oh, I see this was introduced in the last commit. I think it would be best if we only estimate once and increase from our current estimate.